test/node.js

```diff
@@ -5,6 +5,13 @@ import execa from '..';
 
 process.env.PATH = path.join(__dirname, 'fixtures') + path.delimiter + process.env.PATH;
 
+test.beforeEach(t => {
+	t.context.originalArgv = process.execArgv;
+});
+
+test.afterEach(t => {
+	process.execArgv = t.context.originalArgv;
+});
```

Comment:
If using test macros (see comment below), a `try`/`finally` block inside the tests could be used instead of `beforeEach`/`afterEach` hooks.